### PR TITLE
Add bat43 Keyboard

### PR DESCRIPTION
**Pull request by yfuku**
## Description

add bat43 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yfuku** [comment]
@drashna 
Thanks for reviewing it.
I have removed LAYOUT_kc.

File: keyboards/bat43/bat43.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "bat43.h"
+
```
Comment (noroadsleft) [inline keyboards/bat43/bat43.c:18]: Lines 18-50 should be deleted.

File: keyboards/bat43/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 yfuku
```
Comment (noroadsleft) [inline keyboards/bat43/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/bat43/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# bat43
+
+![bat43](https://user-images.githubusercontent.com/30647434/78112239-e9ef8300-7438-11ea-810b-994a75ffa24e.jpg)
+
+A 43-key keyboard.
+
+* Keyboard Maintainer: [yfuku](https://github.com/yfuku)
+* Hardware Supported: The PCBs, controllers supported
```
Comment (noroadsleft) [inline keyboards/bat43/readme.md:8]:
```suggestion
* Hardware Supported: bat43 PCB, Pro Micro
```

File: keyboards/bat43/rules.mk [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
Comment (noroadsleft) [inline keyboards/bat43/rules.mk:12]: As this appears to use a Pro Micro, this should be:

```suggestion
BOOTLOADER = caterina
```

File: keyboards/bat43/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+#define L_SPC LT(_LOWER, KC_SPC)
+#define R_ENT LT(_RAISE, KC_ENT)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT(
+        KC_ESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS,
+        KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_BSPC, KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_RSFT,
+        KC_LCTL, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RCTL,
+                                   KC_LGUI, KC_LANG2,L_SPC,            R_ENT,   KC_LANG1,KC_RALT,
+                                            KC_A,    KC_B,             KC_C,    KC_D,    KC_E
```
Comment (noroadsleft) [inline keyboards/bat43/keymaps/default/keymap.c:34]: I don't actually see where these keys are on the board. Please advise.

Comment (yfuku) [reply keyboards/bat43/keymaps/default/keymap.c:34]: 7E6F7267-EFDD-4756-A66A-6DC37D6CF3D9_1_105_c
Apart from the normal switches, there are five sub switches.
I have attached a photo.

File: keyboards/bat43/bat43.c (hunk 1 above)
Comment (yfuku) [reply keyboards/bat43/bat43.c]: Thank you.
I have removed.

File: keyboards/bat43/keymaps/default/config.h (hunk 2 above)
Comment (yfuku) [reply keyboards/bat43/keymaps/default/config.h]: Thank you.
I have removed.

File: keyboards/bat43/bat43.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                    H5, H6,     H4, D2, D3\
+) { \
+    { A1, A2, A3, A4, A5, A6 }, \
+    { B1, B2, B3, B4, B5, B6 }, \
+    { C1, C2, C3, C4, C5, C6 }, \
+    { D1, D2, D3, D4, D5, D6 }, \
+    { E1, E2, E3, E4, E5, E6 }, \
+    { F1, F2, F3, F4, F5, F6 }, \
+    { G1, G2, G3, G4, G5, G6 }, \
+    { H1, H2, H3, H4, H5, H6}, \
```
Comment (fauxpark) [inline keyboards/bat43/bat43.h:45]: These should be converted to `Kxy` format, to avoid any potential collisions with pin number definitions.

Comment (yfuku) [reply keyboards/bat43/bat43.h]: @fauxpark 
Thanks for the review.
I converted it to the Kxy format.

File: keyboards/bat43/bat43.h [hunk 7]
```diff
@@ -7,7 +7,7 @@
  *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * MERCHANTABILITY or FITNESS FOR PARTICULAR PURPOSE.  See the
```
Comment (fauxpark) [inline keyboards/bat43/bat43.h:10]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```

File: keyboards/bat43/keymaps/default/keymap.c (hunk 5 above)
Comment (noroadsleft) [reply keyboards/bat43/keymaps/default/keymap.c:34]: Ah, I didn't see them on the photos on your Booth page, and I glossed over the 6 tactile switches in the BOM.

That is neat!

File: keyboards/bat43/bat43.h [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+) { \
+    { k00, k01, k02, k03, k04, k05 }, \
+    { k10, k11, k12, k13, k14, k15 }, \
+    { k20, k21, k22, k23, k24, k25 }, \
+    { k30, k31, k32, k33, k34, k35 }, \
+    { k40, k41, k42, k43, k44, k45 }, \
+    { k50, k51, k52, k53, k54, k55 }, \
+    { k60, k61, k62, k63, k64, k65 }, \
+    { k70, k71, k72, k73, k74, k75 }, \
+}
```
Comment (noroadsleft) [inline keyboards/bat43/bat43.h:46]: :heart:

**Comment by noroadsleft** [comment]
Thanks!

**Comment by yfuku** [comment]
Thank you!

**Pull request merged**